pkg/syncer/metadatasyncer.go

```diff
@@ -2382,6 +2382,58 @@ func csiPVUpdated(ctx context.Context, newPv *v1.PersistentVolume, oldPv *v1.Per
 // Vanills k8s and supervisor cluster.
 func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *metadataSyncInformer) {
 	log := logger.GetLogger(ctx)
+	if isPodVMOnStretchSupervisorFSSEnabled {
+		volumeInfo, err := volumeInfoService.GetVolumeInfoForVolumeID(ctx, pv.Spec.CSI.VolumeHandle)
+		if err != nil {
+			log.Errorf("failed to fetch CnsVolumeInfo CR. Error: %+v", err)
+			return
+		}
+		restConfig, err := config.GetConfig()
+		if err != nil {
+			log.Errorf("failed to fetch kubernetes config. Error: %+v", err)
+			return
+		}
+		cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+		if err != nil {
+			log.Errorf("failed to create CNSOperator client. Error: %+v", err)
+			return
+		}
+
+		// Fetch StoragePolicyUsage instance for storageClass associated with the volume.
+		storagePolicyUsageInstanceName := volumeInfo.Spec.StorageClassName + "-" +
+			storagepolicyusagev1alpha1.NameSuffixForPVC
+		storagePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+		err = cnsOperatorClient.Get(ctx, k8stypes.NamespacedName{
+			Namespace: volumeInfo.Spec.Namespace,
+			Name:      storagePolicyUsageInstanceName},
+			storagePolicyUsageCR)
+		if err != nil {
+			log.Errorf("failed to fetch %s instance with name %q from supervisor namespace %q. Error: %+v",
+				storagepolicyusagev1alpha1.CRDSingular, storagePolicyUsageInstanceName,
+				volumeInfo.Spec.Namespace, err)
+			return
+		}
+
+		// Decrease the used capacity in StoragePolicyUsage instance as we are deleting the volume.
+		patchedStoragePolicyUsageCR := storagePolicyUsageCR.DeepCopy()
+		patchedStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Used.Sub(*volumeInfo.Spec.Capacity)
+		err = PatchStoragePolicyUsage(ctx, cnsOperatorClient, storagePolicyUsageCR,
```

Comment:
I wonder if we should check PV deletion policy and only decrease used if the policy is Delete. If this change is in CSI WCP controller, we won't decrease used if the policy is Retain as CSI driver won't get the call. On the other hand, the upstream ResourceQuota mechanism won't check PV deletion policy so this is consistent with that.
Given that, I'm fine merging this.